landoapi/repos.py

```diff
@@ -377,6 +368,10 @@ def ready(self):
                 logger.info("Cloning repo.", extra={"repo": name})
                 r.clone(repo.pull_path)
 
+            # Ensure packages required for automated code formatting are installed.
+            if repo.autoformat_enabled:
+                r.run_mach_bootstrap()
```

Comment:
This my cause some problems if there are changes to `mach bootstrap` that need to be applied after the instance is started (since the `ready` method is only triggered once per restart).

Reply:
For now I think we can run this at repo setup time - we may need to add a mechanism to periodically re-call `mach bootstrap` to keep toolchains up to date. I'd like to land these changes first before optimizing too much. :)